tutorials/sphinx_tuto/tensordict_keys.py

```diff
@@ -0,0 +1,171 @@
+# To access these nested values, we can use tuples of strings. For example
+
+double_nested_a = tensordict["nested", "double_nested", "a"]
+nested_a = tensordict.get(("nested", "a"))
+
+##############################################################################
+# Similarly we can set nested values using tuples of strings
+
+tensordict["nested", "double_nested", "b"] = torch.rand(2, 3)
+tensordict.set(("nested", "b"), torch.rand(2, 3))
+
+print(tensordict)
+
+##############################################################################
+# Iterating over a TensorDict's contents
+# --------------------------------------
+# We can iterate over the keys of a ``TensorDict`` using the ``.keys()`` method.
+
+for key in tensordict.keys():
+    print(key)
+
+##############################################################################
+# By default this will iterate only over the top-level keys in the ``TensorDict``,
+# however it is possible to recursively iterate over all of the keys in the
+# ``TensorDict`` with the keyword argument ``include_nested=True``. This will iterate
+# recursively over all keys in any nested TensorDicts, returning nested keys as tuples
+# of strings.
+
+for key in tensordict.keys(include_nested=True):
+    print(key)
+
+###############################################################################
+# In case you want to only iterate over keys corresponding to ``Tensor`` values, you can
+# additionally specify ``leaves_only=True``.
+
+for key in tensordict.keys(include_nested=True, leaves_only=True):
+    print(key)
+
+################################################################################
+# Much like ``dict``, there are also ``.values`` and ``.items`` methods which accept the
+# same keyword arguments.
+
+for key, value in tensordict.items(include_nested=True):
+    if isinstance(value, TensorDict):
+        print(f"{key} is a TensorDict")
+    else:
+        print(f"{key} is a Tensor")
+
+###############################################################################
+# Checking for existence of a key
+# -------------------------------
+# To check if a key exists in a ``TensorDict``, use the ``in`` operator in conjunction
+# with ``.keys()``.
+#
+# .. note::
+#    Performing ``key in tensordict.keys()`` does efficient ``dict`` lookups of keys
+#    (recursively at each level in the nested case), and so performance is not
+#    negatively impacted when there is a large number of keys in the ``TensorDict``.
+
+assert "a" in tensordict.keys()
+# to check for nested keys, set include_nested=True
+assert ("nested", "a") in tensordict.keys(include_nested=True)
+assert ("nested", "banana") not in tensordict.keys(include_nested=True)
+
+################################################################################
+# Flattening nested keys
+# ----------------------
+# We can flatten a ``TensorDict`` with nested values using the ``.flatten_keys()``
```

Comment:
Let's add that unflattening is super useful for torch.nn.Modules, bc you can get a tensordict structure that looks like your model